docs/website/packages/codegen/generate.md

```diff
@@ -0,0 +1,61 @@
+---
+sidebar_position: 1
+title: Generating
+description: codegen's generate function
+---
+
+The generate function loads FHIR structure definition files, and integrates them into typescript code, which is determined by templates that you prepare. These templates are in [EJS format](https://ejs.co/). The files that generate the definitions can be found at `packages/codegen/definitions/fhir`.
+
+## Basic Example
+
+### Prepare your template
+
+Prepare an EJS formatted template, or multiple templates, and store them in a folder called `/templates` in your project, here is a basic example of a template:
+
+```ejs
+export const ResourceNames = [<% for (const structureDef of fhir.resources) { -%>
+  "<%= structureDef.type -%>",
+<% } -%>];
+```
+
+This template will produce an array of resource names.
+
+### Create a codegen script
+
+We must now create a script for codegen in our `package.json``. here is an example:
```

Comment:
I think adding a script is up to the discretion of the author. We should document it as a command-line tool.

Suggestion:
npx -y @bonfhir/codegen generate --fhir r4b -t ./templates/**/*.ejs -b ./templates -p 'prettier --write %files%' -o ./your-output-path